components/object_store/src/obkv/mod.rs

```diff
@@ -0,0 +1,914 @@
+
+        debug!(
+            "ObkvObjectStore get operation, location:{location}, cost:{}",
+            instant.elapsed().as_millis()
+        );
+        result.box_err().map_err(|source| StoreError::NotFound {
+            path: location.to_string(),
+            source,
+        })
+    }
+
+    async fn get_range(&self, location: &Path, range: Range<usize>) -> Result<Bytes> {
+        let instant = Instant::now();
+
+        let table_name = self.pick_shard_table(location);
+        let meta =
+            self.read_meta(location)
+                .await
+                .box_err()
+                .map_err(|source| StoreError::NotFound {
+                    path: location.to_string(),
+                    source,
+                })?;
+
+        let batch_size = meta.part_size;
+        let key_list = meta.parts;
+        let start_index = range.start / batch_size;
+        let start_offset = range.start % batch_size;
+        let end_index = range.end / batch_size;
+        let end_offset = range.end % batch_size;
+        let mut range_buffer = Vec::with_capacity(range.end - range.start);
+        for (index, key) in key_list
+            .iter()
+            .enumerate()
+            .take(end_index + 1)
+            .skip(start_index)
+        {
+            // let key = &key_list[index];
+            let values = self
+                .client
+                .get(table_name, key.as_bytes())
+                .map_err(|source| StoreError::NotFound {
+                    path: location.to_string(),
+                    source: Box::new(source),
+                })?;
+
+            if let Some(bytes) = values {
+                let mut begin = 0;
+                let mut end = bytes.len();
+                if index == start_index {
+                    begin = start_offset;
+                }
+                if index == end_index {
+                    end = end_offset;
+                }
+                range_buffer.extend_from_slice(&bytes[begin..end]);
+            }
+        }
+
+        debug!(
+            "ObkvObjectStore get_range operation, location:{location}, table:{table_name}, cost:{}",
+            instant.elapsed().as_millis()
+        );
+
+        Ok(range_buffer.into())
+    }
+
+    /// Return the metadata for the specified location
+    async fn head(&self, location: &Path) -> Result<ObjectMeta> {
+        let instant = Instant::now();
+
+        let meta =
+            self.read_meta(location)
+                .await
+                .box_err()
+                .map_err(|source| StoreError::NotFound {
+                    path: location.to_string(),
+                    source,
+                })?;
+
+        debug!(
+            "ObkvObjectStore head operation, location:{location}, cost:{}",
+            instant.elapsed().as_millis()
+        );
+
+        Ok(ObjectMeta {
+            location: (*location).clone(),
+            last_modified: Utc.timestamp_millis_opt(meta.last_modified).unwrap(),
```

Comment:
Shall we avoid `unwrap` here? Maybe we can throw an error here, or use a default value if fails.

Reply:
Ok, I throw an error here.